ONSdigital/rm-notify-gateway
# Move the disable Gov.Notify enabled flag to the NotifyRequestReceiver

# Motivation and Context
Moving the enable/disable flag for integration with Gov.Notify to the Notify Request Receiver. This is the last point before it calls the Gov.Notify service.

This allows us to better test the Notification service in local environments and test environments which do not have Gov.Notify integration enabled as currently the request is stopped at the Rest endpoint so none of the integration with Rabbit is exercised until we get to Preprod.

I've also added some decent error logging when the call to Gov.Notify errors.

# What has changed




# How to test?



1. Run the unit tests and ensure they work
1. Spin up a kubernetes environment and run through the password reset functionality, in the logs you should see output in Notify relating to that request. Also you can check the rabbit queues and ensure that the NotifyRequest queue has received a message.

# Links
https://trello.com/c/tZY5lBL8/1885-18-messages-on-notify-dlq
